### PR TITLE
Implement Unorm10_10_10_2 VertexFormat

### LINKED ISSUE #5110: Missing VertexFormat unorm10_10_10_2 present in other implementations
In https://github.com/gpuweb/gpuweb/pull/4288, a new vertex format was added, but it doesn't seem to be in wgpu.

Is this an oversight, something planned, or intentionally omitted?

If given some pointers (or just an indication that it is desired), I might be able to try adding it myself.

### PR DESCRIPTION
**Pull request by McMackety**
**Connections**
* Fixes https://github.com/gfx-rs/wgpu/issues/5110.
* This PR is dependent on https://github.com/gfx-rs/wgpu/pull/5224 being fixed, as that web-sys version is required, as such this PR does not have a complaint wgpu -> WebGPU backend. (Specifically, the timestamping functionality was disabled)

**Description**
This PR implements the Unorm10_10_10_2 VertexFormat as well as implements the translations required to make it useable on all available backends. 

**Testing**
I manually tested the changes and haven't found any issues, though still need to run it through the updated conformance tests. Additionally, it may be useful to add Unit Tests for this, although currently the different vertex formats are not tested.

**Checklist**

- [X] Run `cargo fmt`.
- [X] Run `cargo clippy`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown` (Failed due to web-sys version)
  - [ ] `--target wasm32-unknown-emscripten` (Failed due to web-sys version)
- [X] Run `cargo xtask test` to run tests.
- [X] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

**Comment by teoxoy** [comment]
@McMackety could you resolve the conflicts?

**Comment by McMackety** [comment]
@teoxoy just resolved them.

File: wgpu-types/src/lib.rs [hunk 1]
```diff
@@ -4942,6 +4942,8 @@ pub enum VertexFormat {
     Float64x3 = 32,
     /// Four double-precision floats (f64). `vec4<f32>` in shaders. Requires [`Features::VERTEX_ATTRIBUTE_64BIT`].
     Float64x4 = 33,
+    /// Three unsigned 10-bit integers and one 2-bit integer, packed into a 32-bit integer (u32). [0, 1024] converted to float [0, 1] `vec4<f32>` in shaders.
+    Unorm10_10_10_2 = 34,
```
Comment (teoxoy) [inline wgpu-types/src/lib.rs]:
```suggestion
    /// Three unsigned 10-bit integers and one 2-bit integer, packed into a 32-bit integer (u32). [0, 1024] converted to float [0, 1] `vec4<f32>` in shaders.
    #[serde(rename = "unorm10-10-10-2")]
    Unorm10_10_10_2 = 34,
```

File: wgpu/src/backend/webgpu.rs [hunk 2]
```diff
@@ -374,6 +374,7 @@ fn map_stencil_state_face(desc: &wgt::StencilFaceState) -> webgpu_sys::GpuStenci
     mapped
 }
 
+
```
Comment (teoxoy) [inline wgpu/src/backend/webgpu.rs]: Could you remove the newlines and add back the line that was removed in this file?

**Review by teoxoy — CHANGES_REQUESTED** [review]

File: wgpu/src/backend/webgpu.rs (hunk 2 above)
Comment (McMackety) [reply wgpu/src/backend/webgpu.rs]: Should be fixed now

File: wgpu/src/backend/webgpu.rs [hunk 3]
```diff
@@ -2517,7 +2518,7 @@ impl crate::context::Context for ContextWebGpu {
         // Not available in gecko yet
         // encoder.pop_debug_group();
     }
-
+
```
Comment (teoxoy) [inline wgpu/src/backend/webgpu.rs]:
```suggestion

```

File: wgpu/src/backend/webgpu.rs [hunk 4]
```diff
@@ -2517,7 +2518,6 @@ impl crate::context::Context for ContextWebGpu {
         // Not available in gecko yet
         // encoder.pop_debug_group();
     }
-
     fn command_encoder_write_timestamp(
```
Comment (teoxoy) [inline wgpu/src/backend/webgpu.rs]:
```suggestion

    fn command_encoder_write_timestamp(
```

**Review by teoxoy — APPROVED** [review]
Looks good, thanks!

File: wgpu-types/src/lib.rs [hunk 5]
```diff
@@ -4942,6 +4942,9 @@ pub enum VertexFormat {
     Float64x3 = 32,
     /// Four double-precision floats (f64). `vec4<f32>` in shaders. Requires [`Features::VERTEX_ATTRIBUTE_64BIT`].
     Float64x4 = 33,
+    /// Three unsigned 10-bit integers and one 2-bit integer, packed into a 32-bit integer (u32). [0, 1024] converted to float [0, 1] `vec4<f32>` in shaders.
+    #[serde(rename = "unorm10-10-10-2")]
```
Comment (teoxoy) [inline wgpu-types/src/lib.rs]:
```suggestion
    #[cfg_attr(feature = "serde", serde(rename = "unorm10-10-10-2"))]
```

**Pull request merged**